galsim/hsm.py

```diff
@@ -661,8 +680,10 @@ def EstimateShear(gal_image, PSF_image, weight=None, badpix=None, sky_var=0.0,
     # prepare inputs to C++ routines: ImageF or ImageD for galaxy, PSF, and ImageI for weight map
     gal_image = _convertImage(gal_image)
     PSF_image = _convertImage(PSF_image)
-    weight = _convertMask(gal_image, weight=weight, badpix=badpix)
     hsmparams = HSMParams.check(hsmparams)
+    if check:
+        _checkWeightAndBadpix(gal_image, weight=weight, badpix=badpix)
```

Comment:
You need to add tests of `check=False`.  (For both functions.)

Reply:
Of course! This was not quite what we discussed in #1264 (we talked about going for a `_FindAdaptiveMomView` there) which is why I first wanted to run this by you before I wrote the tests for the two cases.

Reply:
Sure.  This seems like a reasonable API to me.